Raven.Database/Bundles/Replication/Tasks/IndexReplicationTask.cs

```diff
@@ -44,16 +56,192 @@ public IndexReplicationTask(DocumentDatabase database, HttpRavenRequestFactory h
+                    
+                    foreach (var destination in replicationDestinations)
+                    {
+                        var url = destination.ConnectionStringOptions.Url;
+
+                        var replicatedIndexIds = replicatedSideBySideIndexIds.GetOrAdd(url, new ConcurrentSet<int>());
+                        var indexesToReplicate = sideBySideIndexesToReplicate
+                            .Where(x => replicatedIndexIds.Contains(x.Key) == false).ToList();
+
+                        if (indexesToReplicate.Count == 0)
+                            continue;
+
+                        try
+                        {
+                            ReplicateSideBySideIndexesMultiPut(destination, indexesToReplicate.Select(x => x.Value).ToList());
+                            foreach (var index in indexesToReplicate)
+                            {
+                                replicatedIndexIds.TryAdd(index.Key);
+                            }
+
+                            if (commonReplicatedIndexIds.Count == 0)
+                            {
+                                commonReplicatedIndexIds.UnionWith(replicatedIndexIds);
+                            }
+                            else
+                            {
+                                commonReplicatedIndexIds.IntersectWith(replicatedIndexIds);
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            failedToReplicate = true;
+                            Log.ErrorException("Failed to replicate side by side index to " + destination, e);
+                        }
+                    }
+
+                    if (failedToReplicate)
+                        return;
+
+                    //since we replicated the side by side indexes for all destinations,
+                    //there won't be any need to replicate them again
+                    foreach (var indexId in commonReplicatedIndexIds)
+                    {
+                        IndexToAdd _;
+                        sideBySideIndexesToReplicate.TryRemove(indexId, out _);
+                    }
+                }
+                catch (Exception e)
+                {
+                    Log.ErrorException("Failed to replicate side by side indexes", e);
+                }
+                finally
+                {
+                    interlockedLock.Exit();
+
+                    var newCount = Interlocked.Read(ref sideBySideWorkCounter);
+                    if (currentWorkCount != newCount)
+                    {
+                        //handling a race condition where we get a new side by side index to replicate
+                        //and exiting while thinking that this index will be handled
+                        ReplicateSideBySideIndexes();
+                    }
+                }
+            }, TaskCreationOptions.LongRunning);
```

Comment:
why did you mark it as long running? this forces the thread pool to allocate a new thread for it